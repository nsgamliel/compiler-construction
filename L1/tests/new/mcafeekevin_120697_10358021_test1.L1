(:main
  (:main
    0 0
    call input 0
    rdi <- rax
    mem rsp -8 <- :fib_ret
    call :fib 1
    :fib_ret
    rdi <- rax
    rbx <- rdi
    call print 1
    rdi <- rbx
    mem rsp -8 <- :fib_ret2
    call :fib 1
    :fib_ret2
    rdi <- rax
    call print 1
    return)
  (:fib
    1 0
    rax <- 3
    cjump rdi <= 3 :end
    rax <- 1
    rdi >>= 1
    rsi <- 1
    rcx <- 1
    goto :condition
    :body    
    rdx <- rax
    rax += rcx
    rcx <- rdx
    rsi += 1
    :condition
    cjump rsi < rdi :body
    rax <<= 1
    rax += 1
    :end
    return))