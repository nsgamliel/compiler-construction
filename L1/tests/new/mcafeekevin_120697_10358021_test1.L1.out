433494437
1911095021284458376
